### PR TITLE
A more general support  for QA and summarization (review and feedback needed)

### PR DESCRIPTION
**Pull request by ekurtulus**
With this update, I am aiming for generalizing our training framework. I need feedback and review on the structure that I propose. Accordingly, I will make the codebase fully functional and extend it. Additionally, I added more papers to the docs side.

see [this issue](https://github.com/LAION-AI/Open-Assistant/issues/48)

I think we need support not only for QA but also summarization. In this PR, I wanted to extend the existing datasets, add support for seq2seq models like T5, PolyLoss and few other modifications. Normally, I was also planning to add support for [Sharpness Aware Training](https://github.com/LAION-AI/Open-Assistant/issues/48), but it requires overriding the Huggingface Trainer's __inner_training_loop_ function. I can do that but I am sure if it is something we want. If we decide to do so, I can submit another PR.

Furthermore, if the structure of this commit is approved, I can add more tasks, a script for synthetic data generation through data augmentation, and more.

### CODE REVIEW

**Comment by yk** [comment]
you have some merge conflict markers in your files

**Comment by sanagno** [comment]
Looks good, after the merge! Do you know which models use the SAM optimizer?

File: docs/docs/research/general.md [hunk 1]
```diff
@@ -2,6 +2,43 @@
 
 This page lists research papers that are relevant to the project.
 
+<<<<<<< HEAD
```
Comment (andrewm4894) [inline docs/docs/research/general.md:5]: i think this means you might need to update from upstream/master

**Comment by ekurtulus** [comment]
> Looks good, after the merge! Do you know which models use the SAM optimizer?

None. It is an optimizer you use in conjunction with other optimizers. Here is the [link](https://arxiv.org/pdf/2010.01412) to the paper. There are papers claiming it to be increasing generalization like [this one](https://arxiv.org/pdf/2110.08529.pdf).

**Comment by sanagno** [comment]
Looks great, If you can merge the final conflicts and add a default accuracy metric as before would be great!

**Comment by ekurtulus** [comment]
> @ekurtulus did you close this on purpose?

Yes, I realized that I was not using pre-commit files and not branching properly. While trying to set everything up properly, I had to delete and re-fork the main repository. So, it was closed automatically. I will submit a new PR.

**Comment by ekurtulus** [comment]
Please see [PR 619](https://github.com/LAION-AI/Open-Assistant/pull/619).

**Pull request closed without merging**